src/main/scala/eventstore/EsProjectionsClient.scala

```diff
@@ -0,0 +1,280 @@
+            case status if status.startsWith("Faulted")   => Faulted
+            case other                                    => Other(other)
+          }
+        }
+      }
+      .runWith(Sink.head)
+  }
+
+  private[this] def fetchProjectionData(request: HttpRequest): Future[Option[String]] = {
+    singleRequestWithErrorHandling(request)
+      .mapAsync(1) {
+        case response if response.status == StatusCodes.NotFound => Future.successful(None)
+        case response => Unmarshal(response).to[String].map(Some(_))
+      }
+      .runWith(Sink.head)
+  }
+
+  /**
+   * Fetch the projection's state
+   * @param name the name of the projection
+   * @param partition the name of the partition
+   * @return a String that should be either empty or a valid json object with the current state.
+   */
+  def fetchProjectionState(name: String, partition: Option[String] = None): Future[Option[String]] = {
+    val request = HttpRequest(
+      method = HttpMethods.GET,
+      uri = Uri(fetchProjectionStateUrl(name, partition)),
+      headers = defaultHeaders)
+
+    fetchProjectionData(request)
+  }
+
+  /**
+   * Fetch the projection's result.
+   * It only works for OneTime projections as Continuous one dont provide a result.
+   * @param name the name of the projection
+   * @param partition the name of the partition
+   * @return a String that should be either empty or a valid json object with the projection's result.
+   */
+  def fetchProjectionResult(name: String, partition: Option[String] = None): Future[Option[String]] = {
+    val request = HttpRequest(
+      method = HttpMethods.GET,
+      uri = Uri(fetchProjectionResultUrl(name, partition)),
+      headers = defaultHeaders)
+
+    fetchProjectionData(request)
+  }
+
+  private[this] def executeCommand(name: String, command: String): Future[Unit] = {
+    val request = HttpRequest(
+      method = HttpMethods.POST,
+      uri = Uri(projectionCommandUrl(name, command)),
+      headers = defaultHeaders,
+      entity = HttpEntity(ContentTypes.`application/json`, "{}"))
+
+    singleRequestWithErrorHandling(request)
+      .map {
+        case response if response.status == StatusCodes.NotFound => ()
+        case response if response.status == StatusCodes.OK => ()
+        case response => throw new ProjectionException(s"Received unexpected reponse $response")
+      }
+      .runWith(Sink.head)
+  }
+
+  /**
+   * Blocks for at most #waitDuration for the projection #name to change to the desired status.
+   * @param name the name of the projection to wait on
+   * @param expectedStatus the desired status
+   * @param waitDuration the maximum wait duration
+   * @return true is the expectedStatus was reached, false otherwise
+   */
+  def waitForProjectionStatus(name: String, expectedStatus: ProjectionStatus, waitDuration: FiniteDuration = 10.seconds): Future[Boolean] = {
+    val retryCount = 10
+    val retryDelay = waitDuration / retryCount
+
+    Source(1 to retryCount)
```

Comment:
I actually only used that for tests so i moved it into my tests.
I thought i could be useful in the API but i agree it can be kept out